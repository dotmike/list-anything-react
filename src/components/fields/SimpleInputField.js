import React from "react";
import PropTypes from "prop-types";
import { Form } from "semantic-ui-react";

import InlineError from "../messages/InlineError";

const SimpleInputField = ({
  value,
  type,
  editable,
  showFieldDescription,
  error,
  field,
  onChange
}) => {
  if (editable)
    return (
      <Form.Field error={!!error}>
        <Form.Input
          fluid
          type={type}
          label={field.description}
          value={value}
          onChange={onChange}
          name={field._id}
          placeholder={field.description}
        />
        {error && <InlineError text={error} />}
      </Form.Field>
    );
  if (showFieldDescription)
    return (
      <Form.Field>
        <b>{field.description}:</b> {value}
      </Form.Field>
    );
  return value;
};

SimpleInputField.propTypes = {
  // ownProps
  value: PropTypes.string.isRequired,
  type: PropTypes.string,
  editable: PropTypes.bool,
  showFieldDescription: PropTypes.bool,
  error: PropTypes.string,
  field: PropTypes.shape({
    _id: PropTypes.string,
    description: PropTypes.string
  }),
  onChange: PropTypes.func
};

SimpleInputField.defaultProps = {
  type: "text",
  editable: false,
  showFieldDescription: true,
  error: "",
  field: {},
  onChange: () => {}
};

export default SimpleInputField;
