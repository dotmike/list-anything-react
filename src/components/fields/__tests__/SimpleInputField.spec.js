import SimpleInputField from "../SimpleInputField";

const testData = {
  value: "testando...",
  field: { _id: "1", description: "Testando:" },
  newValue: "test"
};

const setup = (propOverrides = {}) => {
  const props = {
    value: testData.value,
    type: "text",
    error: "",
    editable: true,
    showFieldDescription: true,
    field: testData.field,
    onChange: jest.fn(),
    ...propOverrides
  };

  return {
    props,
    wrapperShallow: wrapperShallow(SimpleInputField, props)
  };
};

describe("SimpleInputField", () => {
  describe("when `editable` is true", () => {
    it("renders correctly", () => {
      const { wrapperShallow: wrapper } = setup({
        editable: true
      });
      expect(wrapper()).toMatchSnapshot();
    });

    it("renders correctly when `error` is passed", () => {
      const { wrapperShallow: wrapper } = setup({
        editable: true,
        error: "Something went wrong..."
      });
      expect(wrapper()).toMatchSnapshot();
    });

    it("calls `onChange` when changing the input value", () => {
      const { wrapperShallow: wrapper, props } = setup({
        editable: true
      });

      wrapper()
        .find("FormInput")
        .simulate("change", { target: { value: testData.newValue } });
      expect(props.onChange).toHaveBeenCalled();
    });
  });

  describe("when `editable` is false", () => {
    it("renders correctly when `showFieldDescription` is true", () => {
      const { wrapperShallow: wrapper } = setup({
        editable: false,
        showFieldDescription: true
      });
      expect(wrapper()).toMatchSnapshot();
    });

    it("renders correctly when `showFieldDescription` is false", () => {
      const { wrapperShallow: wrapper } = setup({
        editable: false,
        showFieldDescription: false
      });
      expect(wrapper()).toMatchSnapshot();
    });
  });
});
