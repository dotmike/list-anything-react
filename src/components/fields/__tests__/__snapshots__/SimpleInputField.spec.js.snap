// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleInputField when \`editable\` is false renders correctly when \`showFieldDescription\` is false 1`] = `"testando..."`;

exports[`SimpleInputField when \`editable\` is false renders correctly when \`showFieldDescription\` is true 1`] = `
<FormField>
  <b>
    Testando:
    :
  </b>
   
  testando...
</FormField>
`;

exports[`SimpleInputField when \`editable\` is true renders correctly 1`] = `
<FormField
  error={false}
>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    label="Testando:"
    name="1"
    onChange={[Function]}
    placeholder="Testando:"
    type="text"
    value="testando..."
  />
</FormField>
`;
